src/hotspot/share/gc/shared/gcLocker.cpp

```diff
@@ -114,6 +136,7 @@ void GCLocker::stall_until_clear() {
 
   // Wait for _needs_gc  to be cleared
   while (needs_gc()) {
+    GCLockerTimingDebugLogger logger("Thread stalled by JNI critical section.");
```

Comment:
If a spurious wakeup occurs, the logger will be instantiated multiple times, this can lead to confusing log msgs, right? If so, I wonder whether it makes sense to extract `logger` out of the while-iteration.

Reply:
Agreed. Same in `GCLocker::jni_lock` below.

It would probably take the form of:

```
if (needs_gc()) {
  GCLockerTracer::inc_stall_count();
  log_debug_jni("Allocation failed. Thread stalled by JNI critical section.");
  GCLockerTimingDebugLogger logger("Thread stalled by JNI critical section.");
  while (needs_gc()) {
    ml.wait();
  }
}
```